objects/src/notes/note_type.rs

```diff
@@ -0,0 +1,71 @@
+use crate::{
+    utils::{
+        format,
+        serde::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable},
+    },
+    Felt, NoteError,
+};
+
+// NOTE TYPE
+// ================================================================================================
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub enum NoteType {
+    /// Notes with this type have only their hash published to the network.
+    OffChain,
+
+    /// Notes with type are shared with the network encrypted.
+    Encrypted,
+
+    /// Notes with this type are fully shared with the network.
+    Public,
+}
```

Comment:
nit: should we set the discriminat values explicitly for each variant?

Reply:
added